app/models/sharing_strategies/work_package_strategy.rb

```diff
@@ -53,6 +53,38 @@ def viewable?
       user.allowed_in_project?(:view_shared_work_packages, @entity.project)
     end
 
+    def share_description(share) # rubocop:disable Metrics/PerceivedComplexity,Metrics/AbcSize
+      return I18n.t("sharing.user_details.invited") if !manageable? && share.principal.invited?
+      return "" if !manageable?
+
+      if share.principal.is_a?(Group)
+        if project_member?(share)
+          I18n.t("sharing.user_details.project_group")
+        else
+          I18n.t("sharing.user_details.not_project_group")
+        end
+
+      elsif group_member?(share)
+        if has_roles_via_group_membership?(share)
+          if project_member?(share)
+            I18n.t("sharing.user_details.additional_privileges_project_or_group")
+          else
+            I18n.t("sharing.user_details.additional_privileges_group")
+          end
+        elsif inherited_project_member?(share)
+          I18n.t("sharing.user_details.additional_privileges_project_or_group")
+        elsif project_member?(share)
+          I18n.t("sharing.user_details.additional_privileges_project")
+        else
+          I18n.t("sharing.user_details.not_project_member")
+        end
+      elsif project_member?(share)
+        I18n.t("sharing.user_details.additional_privileges_project")
+      else
+        I18n.t("sharing.user_details.not_project_member")
+      end
+    end
```

Comment:
I have to concur here. This is already a super complicated tree of conditionals. But the way how it is designed now it is clear to follow a path and see what output it yields to. All optimizations like setting the i18n key to a variable and calling I18n.t with that lead to less readable code here.

Reply:
I don't find that any more readable

Reply:
My point is removal of duplication, so that I don't need to read through prefixes to know that they are the same and there is no typo. I understand not wanting another method especially if it is private and should be separately in private section, then it can be in the same method.
Overall from my perspective having just the meaningful part different in the condition tree makes it easier to read.